yt_dlp/extractor/asobistage.py

```diff
@@ -0,0 +1,160 @@
+        'url': 'https://asobistage.asobistore.jp/event/315passionhour_2022summer/archive/frame',
+        'info_dict': {
+            'id': '315passionhour_2022summer/archive/frame/edff52f2',
+            'title': '315passion_FRAME_only',
+            'live_status': 'was_live',
+            'thumbnail': r're:^https?://[\w.-]+/\w+/\w+',
+        },
+        'params': {
+            'skip_download': True,
+            'ignore_no_formats_error': True,
+        },
+    }, {
+        'url': 'https://asobistage.asobistore.jp/event/idolmaster_idolworld2023_goods/archive/live',
+        'info_dict': {
+            'id': 'idolmaster_idolworld2023_goods/archive/live/3aef7110',
+            'title': 'asobistore_station_1020_serverREC',
+            'live_status': 'was_live',
+            'thumbnail': r're:^https?://[\w.-]+/\w+/\w+',
+        },
+        'params': {
+            'skip_download': True,
+            'ignore_no_formats_error': True,
+        },
+    }, {
+        'url': 'https://asobistage.asobistore.jp/event/ijigenfes_utagassen/player/day1',
+        'only_matching': True,
+    }]
+
+    def _check_login(self, video_id):
+        check_login_json = self._download_json(
+            'https://asobistage-api.asobistore.jp/api/v1/check_login', video_id, expected_status=(400),
+            note='Checking login status', errnote='Unable to check login status')
+        error = traverse_obj(check_login_json, ('payload', 'error_message'), ('error'), get_all=False)
+
+        if not error:
+            return
+        elif error == 'notlogin':
+            self.raise_login_required(metadata_available=False)
+        else:
+            raise ExtractorError(f'Unknown error: {error}')
+
+    def _get_token(self, video_id):
+        return self._search_regex(r'\"([^"]+)\"', self._download_webpage(
+            'https://asobistage-api.asobistore.jp/api/v1/vspf/token', video_id,
+            note='Getting token', errnote='Unable to get token'),
+            name="token")
+
+    def _get_owned_tickets(self, video_id):
+        for url, name in [
+            ('https://asobistage-api.asobistore.jp/api/v1/purchase_history/list', 'ticket purchase history'),
+            ('https://asobistage-api.asobistore.jp/api/v1/serialcode/list', 'redemption history'),
+        ]:
+            for id in traverse_obj(self._download_json(
+                    url, video_id, note=f'Downloading {name}', errnote=f'Unable to download {name}'),
+                    ('payload', 'value', ..., 'digital_product_id', {str})):
+                yield id
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        video_type_name = self._match_valid_url(url).group('type')
+        webpage = self._download_webpage(url, video_id)
+
+        self._check_login(video_id)
+
+        video_type_id, live_status = {
+            'archive': ['archives', 'was_live'],
+            'player': ['broadcasts', 'is_live'],
+        }.get(video_type_name) or [None, None]
```

Comment:
I think I would just
```suggestion
        video_type_id = {
            'archive': 'archives',
            'player': 'broadcasts',
        }.get(video_type_name)
```